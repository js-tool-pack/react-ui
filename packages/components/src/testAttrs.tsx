import { fireEvent, render } from '@testing-library/react';
import type { PropsBase } from '@pkg/shared';
import React from 'react';

export function testAttrs<
  T extends { attrs?: Partial<React.HTMLAttributes<HTMLElement>> },
>(component: React.FC<T>, displayName?: string) {
  const name = displayName || component.displayName || component.name;
  test(name ? `${name} attrs` : `attrs`, () => {
    const onClick = jest.fn();

    const FC = component as React.FC<PropsBase>;
    const { container } = render(
      <FC
        attrs={{
          style: { width: '20px' },
          className: 'foo',
          role: 'note',
          onClick,
        }}
      />,
    );

    expect(container.firstChild).toHaveClass('foo');
    expect(container.firstChild).toHaveAttribute('role', 'note');
    expect(container.firstChild).toHaveStyle({ width: '20px' });

    expect(onClick).not.toBeCalled();
    fireEvent.click(container.firstChild!);
    expect(onClick).toBeCalled();
  });
}
