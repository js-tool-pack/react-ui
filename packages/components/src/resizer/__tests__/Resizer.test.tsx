import { testAttrs } from '~/testAttrs';
import { Resizer } from '..';
import { render } from '@testing-library/react';

describe('Resizer', () => {
  testAttrs(Resizer);

  test('snap', () => {
    const { container } = render(<Resizer>foo bar</Resizer>);
    expect(container.firstChild).toMatchSnapshot();
  });
});
