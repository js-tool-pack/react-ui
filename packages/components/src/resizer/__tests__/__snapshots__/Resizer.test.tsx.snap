// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Resizer snap 1`] = `
<div
  class="t-resizer t-resizer--bottom"
/>
`;
